portality/static/js/doaj.fieldrender.edges.js

```diff
@@ -2320,7 +2320,8 @@ $.extend(true, doaj, {
 
                 var value = false;
                 if (ft === "terms" || ft === "term") {
-                    value = el.attr("data-value");
+                    val = el.attr("data-value");
+                    value = val == "true" ? true : (val == "false" ? false : val);
```

Comment:
These equivalences need to be `===`, but also I want to ponder on the type casting nature of this.  I realise this has a specific use for which this line of code might be the solution, but I'd want to consider making the cast from string to actual value something that is controllable.  Like, what if the string "true" is what we actually want for this filter, or what happens with `0` and `1`, etc.

Reply:
@amdomanska ^ - let me know if you want to talk over it.

Reply:
@richard-jones It doesn't work for only that particular specific case. This is needed because the "data-value" is a string, so we need a cast from string to bool. If the expected value is "Yes", then (just as for the "has the seal" filter), it will keep the string value and use it. So technically speaking, this is the more controllable version of what was previously implemented - as now the bool value is also possible.

You are right, though, that this doesn't take into consideration the `int` value. I've changed that. I double-checked - we use `int` everywhere where it is possible, so there is no risk of converting the string to int when the string is required.

Now the filter can be used for `string`, `int` and `bool` values (tested locally for all the options).

Reply:
Yes, ok, we can go with that for now.  What I was getting at, though, is that the renderer doesn't really know what data type you want to convert that string to, so maybe what we could do is allow the implementer to pass in a coerce function, and then for this one pass in a function which converts "true" and "false" to their boolean equivalents.  That would give you complete generality rather than having to guess the format and all the possible variations.